src/OrleansServiceFabricUtils/ServiceFabricNamingServiceGatewayProvider.cs

```diff
@@ -0,0 +1,372 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Orleans;
+using Orleans.Messaging;
+using Orleans.Runtime;
+using Orleans.Runtime.Configuration;
+
+namespace Microsoft.Orleans.ServiceFabric
+{
+    using System.Fabric;
+    using System.Globalization;
+
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Converters;
+    using Newtonsoft.Json.Linq;
+
+    public class ServiceFabricNamingServiceGatewayProvider : IMembershipTable, IGatewayListProvider
```

Comment:
The name confused me at first. Its both Table and GW Provider, not just GW provider.